map.go

```diff
@@ -27,6 +27,16 @@ func (ms *MapSpec) String() string {
 	return fmt.Sprintf("%s(keySize=%d, valueSize=%d, maxEntries=%d, flags=%d)", ms.Type, ms.KeySize, ms.ValueSize, ms.MaxEntries, ms.Flags)
 }
 
+// Copy returns a copy of the spec.
+func (ms *MapSpec) Copy() *MapSpec {
+	cpy := *ms
+	if ms.InnerMap != nil {
+		innerCopy := *ms.InnerMap
```

Comment:
Why not `.copy()` the inner `MapSpec` too?

Reply:
I think that would cause infinite recursion? I guess I could bail out by returning `nil` if `ms` is `nil`.